src/openforms/payments/base.py

```diff
@@ -32,45 +35,53 @@ class EmptyOptions(JsonSchemaSerializerMixin, serializers.Serializer):
     pass
 
 
-class BasePlugin(AbstractBasePlugin):
+class Options(TypedDict):
+    pass
+
+
+class BasePlugin[OptionsT: Options](AbstractBasePlugin):
     return_method = "GET"
     webhook_method = "POST"
-    configuration_options = EmptyOptions
+    configuration_options: type[serializers.Serializer] = EmptyOptions
 
     # override
 
     def start_payment(
         self,
         request: HttpRequest,
-        payment: "SubmissionPayment",
+        payment: SubmissionPayment,
+        options: OptionsT,
     ) -> PaymentInfo:
         raise NotImplementedError()
 
     def handle_return(
-        self, request: HttpRequest, payment: "SubmissionPayment"
+        self,
+        request: Request,
```

Comment:
changed to DRF request because plugin code is accessing DRF-specific properties that don't exist on `HttpRequest`